central/imageintegration/datastore/datastore_impl_test.go

```diff
@@ -320,13 +334,22 @@ func (suite *ImageIntegrationDataStoreTestSuite) TestEnforcesRemove() {
 }
 
 func (suite *ImageIntegrationDataStoreTestSuite) TestAllowsRemove() {
+	suite.indexer.EXPECT().DeleteImageIntegration(gomock.Any()).Return(nil)
 	integration := suite.storeIntegration("jdgbfdkjh")
 
 	err := suite.datastore.RemoveImageIntegration(suite.hasWriteCtx, integration.GetId())
 	suite.NoError(err, "expected no error trying to write with permissions")
 
+	suite.indexer.EXPECT().DeleteImageIntegration(gomock.Any()).Return(nil)
 	integration = suite.storeIntegration("jdgbfdkjh2")
 
 	err = suite.datastore.RemoveImageIntegration(suite.hasWriteIntegrationsCtx, integration.GetId())
 	suite.NoError(err, "expected no error trying to write with permissions")
 }
+
+func (suite *ImageIntegrationDataStoreTestSuite) TestSearch() {
```

Comment:
This really just checks if `Search` is called. I think it'd be more beneficial to test the search process by creating a real searcher and indexer and then searching for a real element by ClusterID which you have added.

Reply:
Check out https://github.com/stackrox/stackrox/blob/master/pkg/bolthelper/bolt.go#L53 and https://github.com/stackrox/stackrox/blob/master/central/globalindex/bleveindex.go#L81